dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorRecordStore.cs

```diff
@@ -0,0 +1,385 @@
+    public RedisVectorRecordStore(IDatabase database, RedisVectorRecordStoreOptions<TRecord>? options)
+    {
+        // Verify.
+        Verify.NotNull(database);
+
+        // Assign.
+        this._database = database;
+        this._options = options ?? new RedisVectorRecordStoreOptions<TRecord>();
+
+        // Enumerate public properties using configuration or attributes.
+        (PropertyInfo keyProperty, List<PropertyInfo> dataProperties, List<PropertyInfo> vectorProperties) properties;
+        if (this._options.VectorStoreRecordDefinition is not null)
+        {
+            properties = VectorStoreRecordPropertyReader.FindProperties(typeof(TRecord), this._options.VectorStoreRecordDefinition, supportsMultipleVectors: true);
+        }
+        else
+        {
+            properties = VectorStoreRecordPropertyReader.FindProperties(typeof(TRecord), supportsMultipleVectors: true);
+        }
+
+        // Validate property types and store for later use.
+        VectorStoreRecordPropertyReader.VerifyPropertyTypes([properties.keyProperty], s_supportedKeyTypes, "Key");
+        VectorStoreRecordPropertyReader.VerifyPropertyTypes(properties.vectorProperties, s_supportedVectorTypes, "Vector");
+
+        this._keyPropertyInfo = properties.keyProperty;
+        this._keyJsonPropertyName = VectorStoreRecordPropertyReader.GetSerializedPropertyName(this._keyPropertyInfo);
+
+        this._dataPropertyNames = properties
+            .dataProperties
+            .Select(VectorStoreRecordPropertyReader.GetSerializedPropertyName)
+            .ToArray();
+
+        this._dataAndVectorPropertyNames = this._dataPropertyNames
+            .Concat(properties.vectorProperties.Select(VectorStoreRecordPropertyReader.GetSerializedPropertyName))
+            .ToArray();
+
+        // Assign Mapper.
+        if (this._options.MapperType == RedisRecordMapperType.JsonNodeCustomMapper)
+        {
+            if (this._options.JsonNodeCustomMapper is null)
+            {
+                throw new ArgumentException($"The {nameof(RedisVectorRecordStoreOptions<TRecord>.JsonNodeCustomMapper)} option needs to be set if a {nameof(RedisVectorRecordStoreOptions<TRecord>.MapperType)} of {nameof(RedisRecordMapperType.JsonNodeCustomMapper)} has been chosen.", nameof(options));
+            }
+
+            this._mapper = this._options.JsonNodeCustomMapper;
+        }
+        else
+        {
+            this._mapper = new RedisVectorStoreRecordMapper<TRecord>(this._keyJsonPropertyName);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<TRecord> GetAsync(string key, GetRecordOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNullOrWhiteSpace(key);
+
+        // Create Options
+        var collectionName = this.ChooseCollectionName(options?.CollectionName);
+        var maybePrefixedKey = this.PrefixKeyIfNeeded(key, collectionName);
```

Comment:
Do we need to have prefixing functionality? Because `key` and `collectionName` are passed by user. In theory, prefixing can be done on user side, and not only with `collectionName` but with any prefix if needed. So, I would probably skip it here and delegate prefixing logic to the user.

Reply:
I am generally in agreement that we shouldn't do this type of thing in here, but with Redis I expect most customers would want this functionality.  With Redis you can have multiple indexes registered, but a record is only indexed by that index, if the record key is prefixed with the name of the collection.  So prefixing is like adding the record to that "collection".  If prefixing didn't have this direct impact on indexing, I wouldn't have wanted it in here either. As it is, I've made it optional anyway, since I can imagine that some users may want to control this themselves.

Reply:
Makes sense, thanks. If prefixing aspect is common in Redis, then we can support it as well.